apiserver/params/charmhub.go

```diff
@@ -3,75 +3,50 @@
 
 package params
 
-// TODO (hml) 2020-06-17
-// Create actual params.InfoResponse and params.ErrorResponse structs for use here.
 type CharmHubCharmInfoResult struct {
 	Result InfoResponse  `json:"result"`
 	Errors ErrorResponse `json:"errors"`
 }
 
 type InfoResponse struct {
-	Type           string        `json:"type"`
-	ID             string        `json:"id"`
-	Name           string        `json:"name"`
-	Charm          CharmHubCharm `json:"charm,omitempty"`
-	ChannelMap     []ChannelMap  `json:"channel-map"`
-	DefaultRelease ChannelMap    `json:"default-release,omitempty"`
-}
-
-type ChannelMap struct {
-	Channel  Channel  `json:"channel,omitempty"`
-	Revision Revision `json:"revision,omitempty"`
+	Type        string             `json:"type"`
+	ID          string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Publisher   string             `json:"publisher"`
+	Summary     string             `json:"summary"`
+	Series      []string           `json:"series"`
+	StoreURL    string             `json:"store-url"`
+	Charm       CharmHubCharm      `json:"charm,omitempty"`
+	Bundle      CharmHubBundle     `json:"bundle,omitempty"`
```

Comment:
These won't be omitempty unless they're nil and you can't be nil if they're not pointers. We should change them to the following:

```suggestion
	Charm       *CharmHubCharm      `json:"charm,omitempty"`
	Bundle      *CharmHubBundle     `json:"bundle,omitempty"`
```